.github/workflows/lint.yaml

```diff
@@ -0,0 +1,119 @@
+name: lint
+on: [push, pull_request]
+env:
+  GH_ACTION: "TRUE"
+jobs:
+  check-licences:
+    name: Check Licences
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10' 
+      - name: Check licenses
+        shell: bash
+        run: ./scripts/enforce_licence.py
+  generic-nolints:
+    name: Count generic NOLINTs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Count the number of generic NOLINTs
+        shell: bash
+        run: ./scripts/count_generic_nolints.sh
+  build:
+    name: Build for Lint Tasks
+    runs-on: ubuntu-latest
+    container: debian:stable
```

Comment:
Given we should build on Ubuntu too, is this really worth the overhead for linting? I don't know how much inefficiency it actually adds, but still.

Reply:
The "initializing containers" task says it takes 4 seconds, so I don't think it's changing much.

Even though it shouldn't matter for this particular task, I think for consistency it would better to have them all the same unless they explicitly should be different (i.e. have a separate task for building on Ubuntu).

Reply:
> The "initializing containers" task says it takes 4 seconds, so I don't think it's changing much.

I was more meaning what's the overhead on the emulation throughout the run. Essentially what's the time difference with and without that line?

> Even though it shouldn't matter for this particular task, I think for consistency it would better to have them all the same unless they explicitly should be different (i.e. have a separate task for building on Ubuntu).

Yeah fair point. I don't think we need to test Ubuntu AND Debian, and I can see the benefit of testing the deb package on the primary place we release to, just wondering if that should be the edge case. E.g. we test on clang and g++, we could do one of those on Debian and one on Ubuntu for broader testing and minimal overhead/risk of false positives with it being due to the OS not the compiler.

Reply:
>  we could do one of those on Debian and one on Ubuntu for broader testing

Yeah, I was even thinking of including an RPM distro in the build/test workflow if that's something easily supported.

Reply:
Well we've got a probably horribly out of date spec file!